atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1564,6 +1579,22 @@ public void validateLocksOnReadsIfThoroughlySwept() {
                 .isThrownBy(() -> transaction.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL)));
     }
 
+    @Test
+    public void doNotValidateLocksOnCommitIfValidationFlagIsFalseAndOnlyReadNonEmptyValuesOnThoroughlySweptTable() {
+        putCellsInTable(List.of(TEST_CELL), TABLE_SWEPT_THOROUGH);
+
+        long transactionTs = timelockService.getFreshTimestamp();
+        LockImmutableTimestampResponse res = timelockService.lockImmutableTimestamp();
+        Transaction transaction =
+                getSnapshotTransactionWith(timelockService, () -> transactionTs, res, PreCommitConditions.NO_OP, true);
+
+        timelockService.unlock(ImmutableSet.of(res.getLock()));
+
+        transaction.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL));
+
+        assertThatCode(transaction::commit).doesNotThrowAnyException();
+    }
```

Comment:
For completeness: To fail implementations that just set the 'needs validation' flag to false by mistake on reading a present value, we should have a test that reads an empty value and then a non-empty value (and check that we still do validate).

Reply:
ah, good testing case! thanks!